src/Simplex/Messaging/Agent.hs

```diff
@@ -527,7 +532,14 @@ deleteConnectionsAsync_ onSuccess c connIds = case connIds of
 switchConnectionAsync' :: AgentMonad m => AgentClient -> ACorrId -> ConnId -> m ()
 switchConnectionAsync' c corrId connId =
   withStore c (`getConn` connId) >>= \case
-    SomeConn _ DuplexConnection {} -> enqueueCommand c corrId connId Nothing $ AClientCommand $ APC SAEConn SWCH
+    SomeConn _ (DuplexConnection _ rqs@(rq :| _rqs) _) -> case findSwitchedRQ rqs of
+      Nothing -> do
+        let queueNotSwitched = \RcvQueue {switchStatus} -> isNothing switchStatus
```

Comment:
this check seems redundant. You tried to find the queue that switches, found nothing, which means that the primary queue is also not switching - why check it separately?

Reply:
the find is outside of transaction that updates queue status - the second check may be unnecessary because of conn lock, but I thought to keep it as a failsafe